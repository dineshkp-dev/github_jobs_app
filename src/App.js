import Home from './components/Home';
import JobDetails from './components/JobDetails';
import PageNotFound from './components/PageNotFound';
import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import ErrorBoundary from './components/ErrorBoundary';

function App() {
  return (
    <React.Fragment>
      <Router>
        <ErrorBoundary>
          <Switch>
            <Route exact path='/' component={Home}></Route>
            <Route path='/index.html' component={Home}></Route>
            <Route path='/jobdetails/:jobid' component={JobDetails}></Route>
            <Route component={PageNotFound}></Route>
          </Switch>
        </ErrorBoundary>
        {/* <JobDetails /> */}
      </Router>
    </React.Fragment>
  );
}

export default App;
