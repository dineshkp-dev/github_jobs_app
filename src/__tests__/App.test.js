import App from '../App';
import renderer from 'react-test-renderer';

describe('app snapshot test', () => {
  it('renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
