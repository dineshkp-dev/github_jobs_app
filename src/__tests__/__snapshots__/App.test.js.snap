// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot test renders 1`] = `
<div
  className="start"
>
  <div
    className="loading-indicator-container"
  >
    <h3
      className="loading-title"
    >
      Loading...
    </h3>
    <div
      className="loading-text"
    >
      Please wait for job listings to be updated
    </div>
  </div>
</div>
`;
